[18.0][MIG] stock_vertical_lift_empty_tray_check: Migration to 18.0

rlimaeco opened the pull request.

Commit:
[ADD] stock_vertical_lift_empty_tray_check: is the tray empty?

A vertical lift retrieves a tray and places it in front of the
user, and depending on the quantity the user takes from it,
it adapts the pending quantity in the tray. However, because of
errors, it could be that the system thinks the tray is empty
while it is not. With this module, when the system thinks the
tray is empty, while in the step for the release of the tray
the operator is asked explicitly to check if the tray is
actually empty. Depending on his/her answer (yes/no) an inventory
adjustment is created stating the situation.

To activate this optional feature, a new configuration setting
has been added to Inventory > Configuration > Settings, named
'Check Empty Tray'. It is deactivated by default.

Developing decisions:
- The screens shown to the operator are actually wizards, but
  since in the original module (`stock_vertical_lift`) they
  were considered (on the source tree) as views, this has been
  continued here.
- It has been decided, to not change the current workflow of
  the operators, to embed the new check inside the step for
  the 'release'. So, a new screen is shown to ask for the
  visual inspection of whether the tray is empty. In
  order to test this easily, the method `button_release` of
  the module `stock_vertical_lift` has been slightly modified
  so that it always returns. This way we can check easily
  in the unit-tests for the outcome of the intermediate
  screen (i.e. wizard) ─ similarly to how it is done when
  validating a picking that can result in a backorder.

Commit:
[IMP] stock_vertical_lift_empty_tray_check: black, isort, prettier

Commit:
[MIG] stock_vertical_lift_empty_tray_check: Migration to 14.0

oca-travis committed:
[UPD] Update stock_vertical_lift_empty_tray_check.pot

OCA-git-bot committed:
[UPD] README.rst

rousseldenis committed:
[IMP] Update templates with copier

OCA-git-bot committed:
stock_vertical_lift_empty_tray_check 14.0.1.0.1

OCA-git-bot committed:
[UPD] README.rst

etobella committed:
*: Fix pylint

OCA-git-bot committed:
[BOT] post-merge updates

mymage committed:
Added translation using Weblate (Italian)

rlimaeco committed:
[MIG] stock_vertical_lift_empty_tray_check: Migration to 18.0

mmequignon commented:
What is going on here ? @rlimaeco

TDu committed:
[IMP] stock_vertical_lift_empty_tray_check: pre-commit auto fixes

TDu commented:
Fixed the pre-commit commit to not update files from other modules :arrow_up: 
Fixed the settings view :arrow_down:

TDu committed:
[FIX] stock_vertical_lift_empty_tray_check: settings view

Fix the field name which was wrong (no checkbox displayed)
Move the settings in the block dedicated to the verical lift

sebalix committed:
[IMP] stock_vertical_lift_empty_tray_check: bigger title and buttons on empty check wizard

sebalix commented:
Added a new commit to make the title and buttons bigger on wizard pop-up:

## Before
<img width="979" height="205" alt="empty-check-before" src="https://github.com/user-attachments/assets/3f06f987-ca2a-4e52-92f6-bb1636b6c350" />

## After
<img width="976" height="268" alt="empty-check-after" src="https://github.com/user-attachments/assets/a753d140-d540-461d-b20e-12ed99d23f69" />

jbaudoux commented:
/ocabot merge nobump
